### PR TITLE
Ensure the code is compatible with Cython3 + Python3.11

### LINKED ISSUE #1833: Cython3 issue
```
Traceback (most recent call last):
  File "/users/kieffer/workspace-400/pyFAI/build/lib/python3.10/site-packages/pyFAI/test/test_preproc.py", line 149, in test_cython
    self.one_test(cython_preproc)
  File "/users/kieffer/workspace-400/pyFAI/build/lib/python3.10/site-packages/pyFAI/test/test_preproc.py", line 80, in one_test
    res = preproc.preproc(raw)
  File "preproc.pyx", line 640, in pyFAI.ext.preproc.preproc
  File "preproc.pyx", line 463, in pyFAI.ext.preproc.__pyx_fused_cpdef
TypeError: Function call with ambiguous argument types
```
Cython3 is still beta for now ...

### PR DESCRIPTION
**Pull request by kif**
close #1833

### CODE REVIEW

**Comment by kif** [comment]
Overall, those modification present a speed-up of about a 0.2-3s when running the test suite (excluding Qt and OpenCL), tested on python 3.7-3.11 with Cython 0.29.33. Those tests take between 110 and 120s.

**Comment by kif** [comment]
I re-tested all flavors of python interpreter (3.7-3.11) with Cython 3.0b1 and there is a degradation of the performances (+11s in average for the tests run)

File: pyFAI/ext/regrid_common.pxi [hunk 1]
```diff
@@ -41,7 +41,18 @@ __license__ = "MIT"
...
+
+cdef inline double pow2(double x) noexcept nogil:
+    return x*x
+
+cdef inline double pown(double x, unsigned int n) noexcept nogil:
+    cdef double result = 1.0
+    while n:
+        result *= x
+        n-=1
+    return result
```
Comment (t20100) [inline pyFAI/ext/regrid_common.pxi:54]: According to Cython's `cpow` compiler directive, this should not be needed:


Type of a | Type of b | cpow==True | cpow==False
-- | -- | -- | --
C floating point | C integer | Return type is floating point | Return type is floating point

File: pyFAI/ext/inpainting.pyx [hunk 2]
```diff
@@ -44,6 +44,9 @@ from libc.stdint cimport int8_t, uint8_t, int16_t, uint16_t, \
 
 include "bilinear.pxi"
 
+cdef inline double pow2(double x) noexcept nogil:
+    return x*x
+
```
Comment (t20100) [inline pyFAI/ext/inpainting.pyx:49]: Maybe reuse the one from `regrid_common.pxi` or make a dedicated `pxi` file for those, e.g., `math.pxi` or `utils.pxi`

**Review by t20100 — APPROVED** [review]
LGTM

File: pyFAI/ext/inpainting.pyx (hunk 2 above)
Comment (kif) [reply pyFAI/ext/inpainting.pyx]: I did chose this solution ...

**Pull request merged**